src/sentry/taskworker/registry.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+import logging
+import time
+from collections.abc import Mapping
+from datetime import timedelta
+from typing import Any
+from uuid import uuid4
+
+import orjson
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer
+from arroyo.types import Topic as ArroyoTopic
+from google.protobuf.timestamp_pb2 import Timestamp
+from sentry_protos.sentry.v1.taskworker_pb2 import RetryState, TaskActivation
+
+from sentry.conf.types.kafka_definition import Topic
+from sentry.taskworker.retry import FALLBACK_RETRY, Retry
+from sentry.taskworker.task import Task
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options, get_topic_definition
+
+logger = logging.getLogger(__name__)
+
+
+class TaskNamespace:
+    """
+    Task namespaces link topics, config and default retry mechanics together
+    All tasks within a namespace are stored in the same topic and run by shared
+    worker pool.
+    """
+
+    __registered_tasks: dict[str, Task]
+    __producer: KafkaProducer | None = None
```

Comment:
- these annotations should be in `__init__` on the assignments -- putting them here makes it too easy for them to be subtly incorrect or divorced from the actual assignment on a refactor
- [double-underscore does not mean _more_ private](https://youtu.be/IVqLW1NWtPc) -- unless you expect a complex inheritance hierarchy just use single-underscore to denote private